### PR TITLE
Added `react-native run-ios`

### PR DESCRIPTION
**Pull request by frantic**
Works the same way as `react-native run-android`, but targets iOS simulator instead. Under the hood, it uses `xcodebuild` to compile the app and store it in `ios/build` folder, then triggers `instruments` and `simctl` to install and launch the app on simulator.

Since Facebook relies on BUCK to build and run iOS app, we probably won't use `run-ios` internally. That's why I'm putting this as public PR instead of internal diff.

To test this, I hacked global `react-native` script to install react native from my local checkout instead of from npm, cd into the folder and ran `react-native run-ios`.

### CODE REVIEW

**Comment by skevy** [comment]
This is awesome! +1 to opening xcode as little as possible.

**Comment by martinbigio** [comment]
This is awesome! @vjeux and I were talking about this a few days ago :)

File: local-cli/runIOS/parseIOSSimulatorsList.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  name: string;
+  udid: string;
+  version: string;
+}
+
+/**
+ * Parses the output of `xcrun simctl list devices` command
+ */
+function parseIOSSimulatorsList(text: string): Array<IOSSimulatorInfo> {
+  var devices = [];
```
Comment (martinbigio) [inline local-cli/runIOS/parseIOSSimulatorsList.js]: nit: const

File: local-cli/runIOS/parseIOSSimulatorsList.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @flow
+ */
+
```
Comment (martinbigio) [inline local-cli/runIOS/parseIOSSimulatorsList.js]: nit: remove \n

**Comment by dsibiski** [comment]
This is great! :heart_eyes:

**Comment by martinbigio** [comment]
The implementation looks good to me, @frantic feel free to merge it.

**Comment by frantic** [comment]
@facebook-github-bot shipit

**Pull request closed without merging**